File: javascript/6 kyu/Reverse polish notation calculator.js
const calc = (str) => {
    const stack = [];
    const arr = str.split(' ');
    arr.forEach((elm) => {
        let result;
        if (elm === 'sqrt') {
            const nb1 = stack.pop();
            result = Math.sqrt(nb1);
        } else if (isNaN(+elm)) {
            const nb2 = stack.pop();
            const nb1 = stack.pop();
            result = eval(`${nb1} ${elm} ${nb2}`);
        } else {
            result = elm;
        }
        stack.push(result);
    });
    return +stack[0];
};

File: java/6 kyu/Reverse polish notation calculator.java
/**
 *
 * @author Brice
 */

import java.util.Arrays;
import java.util.Stack;

public class Main {

    public static void main(String[] args) {
        double t = Calc.evaluate("1 3 +");
        System.out.println("1 3 +");
        System.out.println(t);
    }

    public static class Calc {
        public static double evaluate(String expr) {
            if (expr.equals("")) return 0.0;
            Stack<Double> polishStack = new Stack<>();
            String[] arrOfStr = expr.split(" ");
            Arrays.stream(arrOfStr)
                    .forEach(s -> {
                        if (s.equals("+") || s.equals("-") || s.equals("*") || s.equals("/")) {
                            Double n2 = polishStack.pop();
                            Double n1 = polishStack.pop();
                            Double result;
                            if (s.equals("+")) {
                                result = n1 + n2;
                            } else if (s.equals("-")) {
                                result = n1 - n2;
                            } else if (s.equals("*")) {
                                result = n1 * n2;
                            } else if (s.equals("/")) {
                                result = n1 / n2;
                            } else {
                                result = 0.00;
                            }
                            polishStack.push(result);
                        } else {
                            Double toPush = Double.parseDouble(s);
                            polishStack.push(toPush);
                        }
                    });
            return (polishStack.pop());
        }
    }
}
